## Filtering Process

The following contains the filtering process used for this project, starting with an unfiltered plink file containing all samples and unfiltered SNPS returned by CAT. This process is similar to the one used by Barria et al. 2018.

We begin with the plink file of all samples called RISG_uf.ped, and remove samples identified by Axiom as low quality, which are contained on a text file named FailedQC.remove

```{bash}
plink --file RISG_uf --chr-set 94 --allow-extra-chr --remove FailedQC.remove --recode --out RISG.passQC
```

Next the individuals were sorted into different plink files by population, this was done using a for loop to expedite the process

```{bash}
for pop in {FI,MV,CT,NR,NEH,GH}; do plink --file RISG.passQC --chr-set 94 --allow-extra-chr --keep-fam $pop.samples --recode --out $pop.uf; done
```

A similar for loop was used for filtering. The plink files were filtered using the following criteria:

Minor Allele Frequency \> 0.05\
Percentage of missingness per SNP \< 0.05\
Percentage of missingness per individual \< 0.05\
Not out of Hardy-Weinberg Equilibrium (p \< 1e-06)\
Non-pathogenic SNPs

```{bash}
for pop in {FI,MV,CT,NR,NEH,GH}; do plink --file $pop.uf --chr-set 94 --allow-extra-chr --maf 0.05 --geno 0.05 --chr 1-10 --mind 0.05 --hwe 1e-6 --recode --out $pop.f; done
```

The filtered files were merged together to create a plink file with all populations and a plink file with just wild populations. The wild populations file will be used for PCA analysis.

```{bash}
plink --file CT.f --chr-set 94 --allow-extra-chr --merge-list Filelist.all --recode --out RISG.all.uf
plink --file CT.f --chr-set 94 --allow-extra-chr --merge-list Filelist.wild --recode --out RISG.wild.uf
```

We only wanted SNPs that were found in all populations after filtering, first SNPs found in all populations were identified and recorded into a text file.

```{bash}
comm -12 <(sort CT.f.map) <(sort FI.f.map) | comm -12 - <(sort GH.f.map) | comm -12 - <(sort NEH.f.map) | comm -12 - <(sort NR.f.map) | comm -12 - <(sort MV.f.map) >> SNPs.txt
```

Next our plink files were filtered to only include common SNPs using the text file created earlier.

```{bash}
plink --file RISG.all.uf --allow-extra-chr --chr-set 94 --extract SNPs.txt --recode --out RISG.all.f
plink --file RISG.wild.uf --allow-extra-chr --chr-set 94 --extract SNPs.txt --recode --out RISG.wild.f
```

Because we are only using the RISG.all file for downstream analysis we will use this file to make a vcf file to be further analyzed in R.

```{bash}
plink --file RISG.all.f --allow-extra-chr --chr-set 94 --recode vcf --out RISG
```
